src/swap/components/SwapSettingsSlippageInput.tsx

```diff
@@ -39,20 +38,15 @@ export function SwapSettingsSlippageInput({
     [updateLifecycleStatus],
   );
 
-  // Handles user input for custom slippage
-  // Parses the input and updates slippage if valid
+  // Handles user input for custom slippage.
+  // Parses the input and updates slippage state.
   const handleSlippageChange = useCallback(
     (newSlippage: string) => {
-      // Empty '' when the input field is cleared.
-      if (newSlippage === '') {
-        setSlippage(0);
-        return;
-      }
+      const parsedSlippage = Number.parseFloat(newSlippage);
```

Comment:
parseFloat('2.') = 2 preventing you from entering decimals

Reply:
I think this is a fast follow update. I belieeve we will need to update `getSwapQuote` and `buildSwapTransaction` as well - this is where number <> string conversation happens again

Reply:
ok, parseFloat(2.2) = 2.2, its just typing the interim step will remove the dot before you type the next number

Reply:
@cpcramer in your update to `getSwapQuote` and `buildSwapTransaction` - can you prevent the APIs from calling when `amount=0`? seems to be an edge case fired off from the component occasionally

Reply:
yep!